rclpy/src/rclpy/_rclpy.c

```diff
@@ -1599,6 +1599,68 @@ rclpy_publish(PyObject * Py_UNUSED(self), PyObject * args)
   Py_RETURN_NONE;
 }
 
+/// Publish a serialized message
+/**
+ * Raises ValueError if pypublisher is not a publisher capsule
+ * Raises RuntimeError if the message cannot be published
+ *
+ * \param[in] pypublisher Capsule pointing to the publisher
+ * \param[in] pymsg serialized message to send
+ * \return NULL
+ */
+static PyObject *
+rclpy_publish_raw(PyObject * Py_UNUSED(self), PyObject * args)
+{
+  PyObject * pypublisher;
+  PyBytesObject * pymsg;
+  char * message;
+  Py_ssize_t pymsg_size;
+
+  if (!PyArg_ParseTuple(args, "OS", &pypublisher, &pymsg)) {
+    return NULL;
+  }
+
+  PyBytes_AsStringAndSize(pymsg, &message, &pymsg_size);
```

Comment:
There's a compiler warning here. I think we can safely cast `pymsg` to `PyObject*`:

```
  PyBytes_AsStringAndSize((PyObject *)pymsg, &message, &pymsg_size);
```